### PR TITLE
Handle FAR for global suppressions in a consistent fashion

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Review by cston — APPROVED** [review]
Compiler changes LGTM.

File: src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs [hunk 1]
```diff
@@ -109,6 +103,22 @@ protected static async Task<ImmutableArray<Document>> FindDocumentsAsync(Project
             }, cancellationToken);
         }
 
+        /// <summary>
+        /// Finds all the documents in the provided project that contain the requested string
+        /// values
```
Comment (ryzngard) [inline src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs:108]: "requested string values" 

What requested string values? This just finds where `ContainsGlobalAttribues` is true

File: src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs [hunk 2]
```diff
@@ -175,49 +181,25 @@ protected static bool IdentifiersMatch(ISyntaxFactsService syntaxFacts, string n
...
-            Func<SyntaxToken, SemanticModel, ValueTask<(bool matched, CandidateReason reason)>> symbolsMatchAsync,
-            CancellationToken cancellationToken)
-        {
-            var findInGlobalSuppressions = ShouldFindReferencesInGlobalSuppressions(symbol, out var docCommentId);
-            return FindReferencesInDocumentUsingIdentifierAsync(
-                identifier, document, semanticModel, symbolsMatchAsync,
-                docCommentId, findInGlobalSuppressions, cancellationToken);
-        }
-
-        [PerformanceSensitive("https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1224834", OftenCompletesSynchronously = true)]
```
Comment (ryzngard) [inline src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs:192]: Unsure why this is no longer perf sensitibe

File: src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs [hunk 3]
```diff
@@ -175,49 +181,25 @@ protected static bool IdentifiersMatch(ISyntaxFactsService syntaxFacts, string n
...
         protected static async ValueTask<ImmutableArray<FinderLocation>> FindReferencesInDocumentUsingIdentifierAsync(
+            ISymbol _,
```
Comment (ryzngard) [inline src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs:185]: Why take a symbol if it's not used?

**Review by ryzngard — APPROVED** [review]
Can we add tests for this? Seems like if we're changing FAR behavior we should have some tests to cover what is consistent

**Comment by ryzngard** [comment]
> Can we add tests for this? Seems like if we're changing FAR behavior we should have some tests to cover what is consistent

Assuming this is a behavior change. With all of the FAR reference changes I'm not 100% sure if this is just a refactoring keeping old behavior. If it is, please add that to the PR description :)

**Comment by CyrusNajmabadi** [comment]
> Can we add tests for this? Seems like if we're changing FAR behavior we should have some tests to cover what is consistent

This area is already extensively tested.  This doesn't change behavior.  It just updates the implementation to follow the general patterns of hte rest of the FAR code and it removes some ugly special casing htat was happening before.  I've pulled this out of a larger set of refactorings i'm making as i would like to keep the large one under control.

**Comment by CyrusNajmabadi** [comment]
> Why take a symbol if it's not used?

It will be used in a followup PR.  If i remove, i need to fixup all callsties, but then i'm going to be adding it back in, so i'll just be undoing that.

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler for a tiny compiler change.

**Pull request merged**